commit 06dbe31e8950b1effb2e0ff98d1162416c406e95
Author: Justin Tee <justin.tee@broadcom.com>
Date:   Thu Dec 12 15:33:04 2024 -0800

    scsi: lpfc: Add handling for LS_RJT reason explanation authentication required
    
    When a LS_RJT is received with reason explanation authentication required,
    current driver logic is to retry the PLOGI up to 48 times.  In the worse
    case scenario, 48 retries can take longer than dev_loss_tmo and if there is
    an RSCN received indicating an authentication requirement change, the
    driver may miss processing it.  Fix by adding logic to specifically handle
    reason explanation authentication required and set the max retry count to 8
    times.
    
    Signed-off-by: Justin Tee <justin.tee@broadcom.com>
    Link: https://lore.kernel.org/r/20241212233309.71356-6-justintee8345@gmail.com
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>

diff --git a/drivers/scsi/lpfc/lpfc_hw.h b/drivers/scsi/lpfc/lpfc_hw.h
index d5c15742f7f2..71d3e60f4b20 100644
--- a/drivers/scsi/lpfc/lpfc_hw.h
+++ b/drivers/scsi/lpfc/lpfc_hw.h
@@ -724,6 +724,7 @@ struct ls_rjt {	/* Structure is in Big Endian format */
 #define LSEXP_OUT_OF_RESOURCE   0x29
 #define LSEXP_CANT_GIVE_DATA    0x2A
 #define LSEXP_REQ_UNSUPPORTED   0x2C
+#define LSEXP_AUTH_REQ          0x48
 #define LSEXP_NO_RSRC_ASSIGN    0x52
 			uint8_t vendorUnique;	/* FC Word 0, bit  0: 7 */
 		} b;